### PR TITLE
feat: vendor slimmed-down versions of the builtin actors

### LINKED ISSUE #2147: Split builtin-actors into versioned crates
**Issue summary**

We depend on the [builtin-actors](https://github.com/filecoin-project/builtin-actors) but only the most recent version used by the Filecoin network is supported at the moment. After discussing with the other Filecoin devs, the Forest team will be reorganizing the structure of the builtin-actor crates and keep them maintained going forward. This issue tracks this work.

- [ ] Create a modified version of `fil_actors_runtime` which isn't tied to a specific network (ie calibnet, mainnet, or devnet). The network specific parts should be explicitly named or moved to a separate crate. The crate itself could be renamed to, say, `fil_actors_runtime_common` to avoid name clashes.
- [ ] Create v8 versions of each actor. For example, the v8 account actor should live in `builtin-actors/actors/account_v8` and the crate name should include v8.
- [ ] Create v9 versions of each actor. For example, the v9 account actor should live in `builtin-actors/actors/account_v9` and the crate name should include v9.
- [ ] Split actor crate in two: One for the states and state methods, one for the smart contract code.
- [ ] Update the CI to automatically test the v8 and v9 actors.

**Other information and links**

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- [x] `runtime_v8`
- [x] `account_v8`
- [X] `cron_v8`
- [x] `init_v8`
- [x] `market_v8` 
- [X] `miner_v8`
- [X] `multisig_v8`
- [x] `paych_v8` 
- [x] `power_v8`
- [x] `reward_v8`
- [X] `system_v8`
- [X] `verifreg_v8`
- [x] `runtime_v9`
- [x] `account_v9`
- [X] `cron_v9`
- [x] `init_v9`
- [x] `market_v9` 
- [X] `miner_v9`
- [X] `multisig_v9`
- [x] `paych_v9`
- [x] `power_v9`
- [x] `reward_v9`
- [X] `system_v9`
- [X] `verifreg_v9`

We'll be vendoring the builtin-actors without any of the smart-contract code. Each actor version (eg v8 and v9) will live in its own crate. As more and more functionality will become available through WASM, hopefully the need for these crates will eventually disappear.

**Reference issue to close (if applicable)**

Closes #2147

**Other information and links**

Discussion on builtin-actors: https://github.com/filecoin-project/builtin-actors/issues/194

### CODE REVIEW

**Comment by lemmih** [comment]
@sudo-shashank @tyshko5 Each actor can be worked on in parallel (well, a few actors depend on other actors but most of the work can be done in parallel). If you start working on an actor, please check it off the list. Feel free to push directly into my branch.

File: blockchain/blocks/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ edition = "2021"
 base64.workspace            = true
 byteorder.workspace         = true
 cid.workspace               = true
-derive_builder              = "0.11"
+derive_builder              = { workspace = true }
```
Comment (LesnyRumcajs) [inline blockchain/blocks/Cargo.toml:11]:
```suggestion
derive_builder.workspace    = true
```

File: encoding/Cargo.toml [hunk 2]
```diff
@@ -12,5 +12,5 @@ blake2b_simd.workspace       = true
 cs_serde_bytes.workspace     = true
 serde                        = { workspace = true, features = ["derive"] }
 serde_ipld_dagcbor.workspace = true
-serde_repr                   = "0.1"
+serde_repr                   = { workspace = true }
```
Comment (LesnyRumcajs) [inline encoding/Cargo.toml:15]:
```suggestion
serde_repr.workspace         = true
```

File: ipld/Cargo.toml [hunk 3]
```diff
@@ -15,7 +15,7 @@ cid.workspace               = true
 forest_encoding.workspace   = true
 fvm_ipld_encoding.workspace = true
 fvm_shared                  = { workspace = true, default-features = false }
-indexmap                    = { version = "1.9", features = ["serde-1"] }
+indexmap                    = { workspace = true }
```
Comment (LesnyRumcajs) [inline ipld/Cargo.toml:18]:
```suggestion
indexmap.workspace          = true
```

File: ipld/legacy_amt/Cargo.toml [hunk 4]
```diff
@@ -16,7 +16,7 @@ forest_encoding.workspace     = true
 forest_utils.workspace        = true
 fvm_ipld_blockstore.workspace = true
 fvm_ipld_encoding.workspace   = true
-itertools                     = "0.10"
+itertools                     = { workspace = true }
```
Comment (LesnyRumcajs) [inline ipld/legacy_amt/Cargo.toml:19]:
```suggestion
itertools.workspace           = true
```

**Review by LesnyRumcajs — APPROVED** [review]
LGTM, we should probably link the https://github.com/ChainSafe/fil-actor-states somewhere in the README.

**Review by elmattic — APPROVED** [review]

**Pull request merged**